invoice/src/main/java/org/killbill/billing/invoice/notification/DefaultNextBillingDatePoster.java

```diff
@@ -111,7 +112,7 @@ private void insertNextBillingFromTransactionInternal(final EntitySqlDaoWrapperF
             if (existingNotificationForEffectiveDate == null) {
                 log.info("Queuing next billing date notification at {} for subscriptionId {}", futureNotificationTime.toString(), JOINER.join(subscriptionIds));
 
-                final NextBillingDateNotificationKey newNotificationEvent = new NextBillingDateNotificationKey(null, subscriptionIds, targetDate, isDryRunForInvoiceNotification);
+                final NextBillingDateNotificationKey newNotificationEvent = new NextBillingDateNotificationKey(null, subscriptionIds, targetDate, isDryRunForInvoiceNotification, isRescheduled);
```

Comment:
⚠️  The issue here is that if we end up inserting a new notification, all is good, but if not, how do we handle the `isRescheduled` flag?

Reply:
See https://github.com/killbill/killbill/pull/923/files#diff-9f199d591e4f5b2d3ba23e4b87e150d4R532, where I add a notification at `rescheduleDate` for each subscription id.